lgc/patch/PatchBufferOp.cpp

```diff
@@ -1400,14 +1400,26 @@ Value *PatchBufferOp::replaceLoadStore(Instruction &inst) {
   Value *storeValue = nullptr;
   if (!isLoad) {
     storeValue = storeInst->getValueOperand();
-
-    if (storeValue->getType()->isPointerTy()) {
+    Type *storeTy = storeValue->getType();
+    if (storeTy->isPointerTy()) {
```

Comment:
I think the logic is expected to change like this:

```
if ((storeTy->isArrayTy()) {
  // Your new codes
} else {
  // Old codes
  if (storeTy->isPointerTy())
    XXX;

  YYY;
}
```

Reply:
Fixed.